includes/db/Database.php

```diff
@@ -3519,21 +3521,24 @@ protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
 	 * at the rate defined in self::QUERY_SAMPLE_RATE.
 	 *
 	 * @param string $sql the query
+	 * @param ResultWrapper|resource $ret database results
 	 * @param string $fname the function name
 	 * @param bool $isMaster is this against the master
 	 * @return void
 	 */
-	protected function logSql( $sql, $fname, $elapsedTime, $isMaster ) {
+	protected function logSql( $sql, $ret, $fname, $elapsedTime, $isMaster ) {
 		global $wgDBcluster;
 
 		if ($this->getSampler()->shouldSample()) {
 			$this->getWikiaLogger()->info( "SQL $sql", [
 				'method'      => $fname,
 				'elapsed'     => $elapsedTime,
+				'num_rows'    => ( $ret instanceof ResultWrapper ? $ret->numRows() : mysql_num_rows( $ret ) ),
 				'cluster'     => $wgDBcluster,
 				'server'      => $this->mServer,
 				'server_role' => $isMaster ? 'master' : 'slave',
 				'db_name'     => $this->mDBname,
+				'exception'   => new Exception(), // log the backtrace
```

Comment:
we keep adding exception here and there, maybe we should add it by default in the logger?